### PR TITLE
Failed scan closed

### PR DESCRIPTION
**Pull request by ivakegg**

### CODE REVIEW

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java
Comment (ddanielr) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java:1]: I'm not sure why these changes are showing up again. The file contents already match this state in 45a8a66161b00f512bc6dd2563f8c2df4ca6e32c

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java [hunk 1]
```diff
@@ -90,25 +88,24 @@ class ScanDataSource implements DataSource {
...
-      if (fileManager != null) {
-        tablet.getScanMetrics().decrementOpenFiles(fileManager.getNumOpenFiles());
-        fileManager.releaseOpenFiles(false);
+      try {
+        returnIterators();
+      } finally {
+        try {
+          if (fileManager != null) {
+            tablet.getScanMetrics().decrementOpenFiles(fileManager.getNumOpenFiles());
+            fileManager.releaseOpenFiles(false);
```
Comment (ddanielr) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java:101]: releaseOpenFiles calls releaseReaders which can throw an IllegalArgumentException 

https://github.com/ivakegg/accumulo/blob/14dd12c3a97f44cd6c0273552ce8bc3b5439cd20/server/tserver/src/main/java/org/apache/accumulo/tserver/FileManager.java#L360

releaseOpenFiles is a synchronized method so it could get moved into returnIterators as releasing and decrementing the openFiles are actions performed by both close() and getNewDataSource()

Comment (ivakegg) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java:101]: one of them calls releaseOpenFiles(false) and the other releaseOpenFiles(sawErrors).  Also note that this is within a try/finally block.

Comment (ddanielr) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java:101]: The only difference is that `close()` sets the filemanager to null once it's done 
```
          if (fileManager != null) {
            tablet.getScanMetrics().decrementOpenFiles(fileManager.getNumOpenFiles());
            fileManager.releaseOpenFiles(false);
```

Comment (ivakegg) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java:101]: Also one of them clears fileManager and the other does not.
